packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -142,9 +142,11 @@ class UserSessionManager implements IUserSessionManager {
       };
     }
 
-    state.session.sessionInfo.value = this.isPersistenceEnabledForStorageEntry('sessionInfo')
-      ? (sessionInfo as SessionInfo)
-      : DEFAULT_USER_SESSION_VALUES.sessionInfo;
+    state.session.sessionInfo.value =
```

Comment:
At this point, the `sessionInfo` variable has already taken into account the configured load options in Ln 139 and 140. 
The load option values take precedence.

Ln 140 says that auto-tracking should be enabled only if the load option flag is set and in the storage manual tracking is not configured.

Are we trying to solve a different problem where auto tracking is configured to be off so we do not persist anything in the storage (by using the default value `{}`)?

Reply:
The primary problem is with existing session info in the storage, when autoTrack is set to false via loadOptions, we are still seeing sessionId in the event payload. The reason behind this is we are not clearing the sessionInfo.
So, by setting it to default value we are resolving both the problems.

Reply:
At what point in the code that bug is introduced? 
The logic seems clear as I explained above.

Reply:
Not sure

Reply:
What do you mean by not sure? Have you not debugged the issue line by line using breakpoints?